### PR TITLE
Set catalog on `SchemaDeployer` to overwrite the default `hive_metastore`

### LINKED ISSUE #294: [FEATURE] Set catalog on `SchemaDeployer`
### Description

Set the catalog on `SchemaDeployer` with a default of `hive_metastore: https://github.com/databrickslabs/lsql/blob/8fd3b9e1b89b086c455881e98ac967e1194ecd88/src/databricks/labs/lsql/deployment.py#L15

Then use the catalog in the queries in the `SchemaDeployer`'s methods.

### PR DESCRIPTION
**Pull request by JCZuurmond**
Set catalog on `SchemaDeployer` to overwrite the default `hive_metastore`

### Linked issues

Resolves #294
Needs #280 (tech debt to tackle later)
Progresses #278
Requires #287 for the CI to pass

### CODE REVIEW

File: src/databricks/labs/lsql/deployment.py [hunk 1]
```diff
@@ -1,46 +1,53 @@
...
-    def __init__(self, sql_backend: SqlBackend, inventory_schema: str, mod: Any):
+    """Deploy schema, tables, and views for a given inventory schema."""
+
+    def __init__(
+        self,
+        sql_backend: SqlBackend,
+        inventory_schema: str,
+        mod: Any,
+        *,
+        inventory_catalog: str = "hive_metastore",
```
Comment (nfx) [inline src/databricks/labs/lsql/deployment.py:19]:
```suggestion
        catalog: str = "hive_metastore",
```
let's use new naming conventions for new feat

File: src/databricks/labs/lsql/deployment.py [hunk 2]
```diff
@@ -1,46 +1,53 @@
...
 
 
 class SchemaDeployer:
-    def __init__(self, sql_backend: SqlBackend, inventory_schema: str, mod: Any):
+    """Deploy schema, tables, and views for a given inventory schema."""
+
+    def __init__(
+        self,
+        sql_backend: SqlBackend,
+        inventory_schema: str,
```
[the remark this answers is not in the document]
Comment (nfx) [reply src/databricks/labs/lsql/deployment.py:16]: it's a small change - perhaps it's easier to do it in this one

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/lsql/core.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ class StatementExecutionExt:
     megabytes or gigabytes of data serialized in Apache Arrow format, and low result fetching latency, should use
     the stateful Databricks SQL Connector for Python."""
 
-    def __init__(  # pylint: disable=too-many-arguments
+    def __init__(  # pylint: disable=too-many-arguments,too-many-positional-arguments
```
Comment (JCZuurmond) [inline src/databricks/labs/lsql/core.py:143]: No lint CI fails on this line, which was required by the linter

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**